libminifi/include/ResourceClaim.h

```diff
@@ -112,7 +122,7 @@ class ResourceClaim : public std::enable_shared_from_this<ResourceClaim> {
   // Full path to the content
   std::string _contentFullPath;
 
-  std::shared_ptr<core::StreamManager<ResourceClaim>> claim_manager_;
```

Comment:
hmmm why are we creating a non-owning reference here? We could potentially lose data in some routes. Seems worrisome. I'll do some testing to validate.